### ISSUE TITLE
Try deeper models

### ISSUE THREAD

**Issue by egpbos** [spokenlanguage/platalea#32]
In lunar-serenity-26, dark-serenity-27 and worthy-sunset-28, we notice that dropout may help slightly to get validation and training loss closer together, but at a cost to performance. Although it might be worth seeing how dropout models fare when trained for more epochs (only 32 in the mentioned runs), one other promising avenue may be to look into using deeper and wider models, as [suggested here](﻿https://towardsdatascience.com/this-is-how-to-train-better-transformer-models-d54191299978) (in turn based on [this article](https://arxiv.org/pdf/2002.11794.pdf)).

Compared to that paper, we are on the low end of both width (we used d_model = 256 in the above runs) and depth (4 now).

Glancing over their paper, it seems they are getting diminishing returns after d_model = 768 and depth = 12.

Their data and tasks are vastly different, they use multi-million word text sets and hence also completely different tasks.

Still, we should look into scaling up these parameters and seeing what it does.

**Comment by egpbos** [comment]
Run [upbeat-wind-33](https://wandb.ai/spokenlanguage/platalea_transformer/runs/cdu96e8i) has d_model = 512 (instead of 256 in previous runs).

It also uses the new CPU-scoring option, which reloads the saved model from disk into the CPU instead of the GPU to run the `score` function. This avoids the issue of going over the GPU's memory limit in the score step (at the end of each epoch), which became a critical issue with these parameters (and was actually the reason for choosing these lower than standard parameters in the first place).

**Comment by egpbos** [comment]
Run brisk-haze-32 had the same parameters, by the way, but was stopped because I was still testing things.

One thing we can already conclude from the first steps: this model takes about 2.5 times longer in wall clock time per epoch. A big chunk of this is actually the CPU-scoring; that part actually takes the same amount of time as all the training steps in the epoch, whereas on GPU it takes a few percent.

**Comment by cwmeijer** [comment]
We tried deeper models up to 10 layers (from the 4 we used to use). The effect isn't super clear. It's definitely not making a big difference in terms of performance.
See https://wandb.ai/spokenlanguage/platalea_transformer/reports/Feb-23-Project-Update-grid-search-on-trafo-layers-and-heads--Vmlldzo0ODQ5MzE for a report.
